### ISSUE TITLE
sync git folder (and similar folders with internal consistency constrains, e.g. database using file storage)

### ISSUE THREAD

**Issue by andrewufrank** [syncthing/syncthing#7627]
I use git and allow synthing to sync the folders I work on to have the same work environment on different devices, e.g. when I move from work to home. I understand that I can ignore the git files from sync and then would have to achieve the update manually with git tools, but I prefer to sync. 

Occasionally, I find git repositories corrupted; I assume because syncthing does not understand the consistency rules git enforces in its updates and combines updated files level from different changes, which results in an inconsistent git state (for syncthing some files are found in conflict and marked, but the directory is inconsistent for git). For this reasons, syncing git folders (independent of which tool does the sync) is not advisable. The same warning applies to a number of other programs which store data in files in a directory and maintain some consistency among them. 

Would it be possible to declare directories (e.g. the git directories) to be synced such that "consistent" states would be synced between nodes. Consistent state for the directory would mean: all files in the directory would synced together and conflicts would be detected on the directory level not the file as now). The guarantee is that the result of the sync is a state of the directory which existed on some of the host (and the state of directories on other hosts would be copied and marked as 'in conflict'); files from different states of the directory would not arrive in one directory.

Such an optional policy would be useful for more than just git, but also for mailboxes, databases like the syncthing index etc. 

Implementation idea: a mark could be set on the directory (with a hidden file e.g. `.syncDirWhole` added manually). Such directories would be synced as wholes, in one transaction - either all done or not at all - and conflicts detected marked at the directory.

**Comment by imsodin** [comment]
Duplicate of https://github.com/syncthing/syncthing/issues/4608

tl;dr: This is out-of-scope for Syncthing.
